### PR TITLE
tx-generator: tx-generator.service for Plutus

### PR DESCRIPTION
**Pull request by MarcFontaine**
tx-generator: Add Plutus support to JSON config.
* Add payMode.PayToScript and spendMode.SpendScript
* JSON-configurable execution units.
* Use protocolParamPrices in tx-generator.
* Better handling of collaterals in tx-generator.
* Update tx-generator-service.nix and hard-coded test-scripts.
* TODO: configurable ScriptRedemer and ScriptDatum.

### CODE REVIEW

File: nix/nixos/tx-generator-service.nix [hunk 1]
```diff
@@ -1,5 +1,7 @@
 pkgs:
 let
+  plutus = true;
+  plutusScriptFile = "../../../bench/script/sum1ToN.plutus";
```
Comment (MarcFontaine) [inline nix/nixos/tx-generator-service.nix:4]: Here the Plutus settings are hardcoded.
Instead proper parameters from the nix-service definition should be used instead.

**Review by deepfire — APPROVED** [review]
Looks good, thank you Marc!

**Pull request merged**